File: Sismique/ClientApp/app/components/shared/category.ts
import { ImageCategory, IImageCategoryProperties } from "./imagecategory";

export interface ICategoryProperties {
    id: number;
    name: string;
    order: number;
    overviewCategory: boolean;
    drawingCategory: boolean;
    visible: boolean;
    imageCategories: IImageCategoryProperties[];
}

export class Category {
    id: number;
    name: string;
    order: number;
    overviewCategory: boolean;
    drawingCategory: boolean;
    visible: boolean;
    imageCategories: ImageCategory[];

    deserialize(input: ICategoryProperties): Category {
        this.id = input.id;
        this.name = input.name;
        this.order = input.order;
        this.overviewCategory = input.overviewCategory;
        this.drawingCategory = input.drawingCategory;
        this.visible = input.visible;

        this.imageCategories = [];
        if (input.imageCategories != null) {
            for (let ic of input.imageCategories) {
                const imageCategories = new ImageCategory();
                imageCategories.deserialize(ic);
                this.imageCategories.push(imageCategories);
            }
        }
        return this;
    }
}

File: Sismique/Models/Category.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace ARIO.Models
{
    public class Category
    {
        public int ID { get; set; }

        // Name as reported by the classifier
        [Required] public string ClassifierName { get; set; }

        [Required] public string Name { get; set; }

        [Required] public int Order { get; set; }

        [Required] public bool OverviewCategory { get; set; }

        [Required] public bool DrawingCategory { get; set; }

        [Required] public bool Visible { get; set; }

        public int CategoryTypeId { get; set; }

        public CategoryType CategoryType { get; set; }

        public ICollection<ImageCategory> ImageCategories { get; } = new List<ImageCategory>();
    }
}